network/betweenness.go

```diff
@@ -32,9 +32,61 @@ func Betweenness(g graph.Graph) map[int]float64 {
 	// Also note special case for sparse networks:
 	// http://wwwold.iit.cnr.it/staff/marco.pellegrini/papiri/asonam-final.pdf
 
-	var (
-		cb = make(map[int]float64)
+	cb := make(map[int]float64)
+	brandes(g, func(s graph.Node, stack internal.NodeStack, p map[int][]graph.Node, delta, sigma map[int]float64) {
+		for stack.Len() != 0 {
+			w := stack.Pop()
+			for _, v := range p[w.ID()] {
+				delta[v.ID()] += sigma[v.ID()] / sigma[w.ID()] * (1 + delta[w.ID()])
+			}
+			if w.ID() != s.ID() {
+				if d := delta[w.ID()]; d != 0 {
+					cb[w.ID()] += d
+				}
+			}
+		}
+	})
+	return cb
+}
+
+// EdgeBetweenness returns the non-zero betweenness centrality for edges in the unweighted graph g.
+//
+//  C_B(q(u, v)) = \sum_{s ≠ t, u ≠ v ∈ V} (\sigma_{st}(q) / \sigma_{st})
```

Comment:
A little bit of nitpicking. `q(u,v)` is the edge betweeen u and v, but I don't think we use q to denote edges anywhere else. Also, the formula's notation is strange. u and v appear as independent "variables" on the left-hand side but also as if they are summed over. I would write:
```
// EdgeBetweenness returns the non-zero betweenness centrality for edges in the unweighted graph g.
// For an edge e the centrality is C_B is computed as
//
//  C_B(e) = \sum_{s ≠ t ∈ V} (\sigma_{st}(e) / \sigma_{st}),
//
// where \sigma_{st} and \sigma_{st}(e) are the number of shortest paths from s to t,
// and the subset of those paths containing e, respectively.
//
// If g is undirected, edges are retained such that u.ID < v.ID where u and v are the nodes of e.
```
(or something like that). It is not necessary to mention u and v until the very end.